src/mqt/problemsolver/partialcompiler/Partial_QAOA.py

```diff
@@ -0,0 +1,192 @@
+import numpy as np
+from qiskit import QuantumCircuit, transpile
+from qiskit.circuit import Parameter
+from qiskit.providers.fake_provider import FakeManila, FakeMontreal, FakeWashington
+
+P_SAMPLE_TWO_QUBIT_GATE = 0.5
+IBM_GATES = ["cx", "sx", "x", "rz"]
+
+
+class Partial_QAOA:
+    def __init__(self, num_qubits: int, repetitions: int = 1):
+        """
+        Creates a QAOA problem instance with a random number of known/offline edges and a random number of unknown/online edges.
+
+        :param num_qubits: Number of qubits in the problem instance
+        :param repetitions: Number of repetitions of the problem and mixer unitaries
+        """
+        self.num_qubits = num_qubits
+        self.repetitions = repetitions
+        self.offline_time_edges = []
+
+        # add offline edges
+        for i in range(1, num_qubits):
+            if np.random.random() < P_SAMPLE_TWO_QUBIT_GATE:
+                self.offline_time_edges.append((0, i))
```

Comment:
The decision to have all offline edges share a common index is on purpose, right?

Reply:
Yes. The motivation was to have some offline edges and to make sure, that at least one qubit is frequently interacted with such that the pre-determined mapping can exploit that.

Reply:
Makes sense 👍🏼